### ISSUE TITLE
Package conflicts with other wine packages

### ISSUE THREAD

**Issue by BonusPlay** [M-Reimer/wine-lol#1]
Forcing a decision between wine-lol and other wine instances is really bad. e.g. [wine-osu](https://aur.archlinux.org/packages/wine-osu/) package doesn't do that. Maybe this could be changed?

**Comment by M-Reimer** [comment]
Maybe optional and disabled by default.
But I would prefer to still have one wine in my system.
After all this is just standard wine-staging with one additional patch.

**Comment by BonusPlay** [comment]
But it requires to build wine on your machine every update. I think everyone can spend additional 200MB (tbh I have no idea how much space wine needas) for another wine install, instead of lengthy compilation each update.

**Comment by M-Reimer** [comment]
You'll need this from time to time anyway.
If glibc is updated, you **have to** rebuild wine-lol-glibc. There is no version difference allowed as the wine build system didn't make it easily possible to change the dynamic linker path.
And my guess is that this is not the last time that wine needs patching. With the last wine-staging versions *every single time* there was something which fixed stuff on League.
It's just bad that it will take some time now until a fix is found by the Wine developers which can be staged. As soon as this happens, "wine-lol" will be depracated again until the next bigger problem occurs.

**Comment by M-Reimer** [comment]
Placing this below "opt" would be just a minor change (move the prefix to a variable that can be changed if someone wants to).

Maybe I'll add this as an optional feature.

But I would not recommend anyone to do this. wine-lol is meant to go as soon as possible. As soon as wine-staging has the required fixes again, it is time to replace wine-lol back with wine-staging.

If you didn't install it to somewhere else, you just replace the package in this case and everything just continues to work as usual.

Same if another problem should come in future. In this case I could try to get "bleeding edge" patches into "wine-lol" again and users could again replace their wine-staging with wine-lol.

So in other words: The desired use case is to replace wine-staging for as long as needed and as soon as possible get rid of wine-lol again for as long as it takes for another problem to occur where a "patched wine-staging" can help to bridge the time it takes for the fixes to land in wine-staging again.

**Comment by BonusPlay** [comment]
What if wine-staging releases 4.9 which fixes league issue? I don't want to switch packages non-stop/compile wine on every update. Keeping wine-lol as separate wine instance has no disadvantages, but does have advantages. Whenever league breaks wine-staging again, you can push bleeding edge update to wine-lol.

**Comment by M-Reimer** [comment]
I don't want to maintain a League Wine forever. As soon as wine staging takes over, my job is done for as long as it takes for the next problem to come.

**Comment by M-Reimer** [comment]
And I did forget to mention that this whole "patched glibc" stuff is hacky and will need some maintenance in future. So it would be best to get rid of this construction as soon as possible.

**Comment by M-Reimer** [comment]
Will keep this Issue open until it gets clearer how this is fixed in the end.
With this glibc hack problem there is quite some room for speculation about when and how this fix ends in wine-staging.

Just for the case, and to learn a bit more about the Wine build system, my next step (next weekend) will be to try to get the Wine binaries linked to the "wine-lol-glibc" dynamic linker. If I manange to do this, "wine-lol-glibc" could be a different version than the glibc on the system.

**Comment by M-Reimer** [comment]
I thought a bit more about this topic.
Actually the case "wine-staging takes over" can be really easily modeled in an PKGBUILD.
I would just push an update which completely drops the whole "wine-lol" build and pulls wine-staging as a dependency. Then a simple symlink from /usr/bin/wine to /usr/bin/wine-lol is installed by my package.

If a new patch is needed in future, I could again make my PKGBUILD build a patched wine and keep this as long as needed before replacing with a "symlink creating package" again.

Advantages of this approach:
- No need to repeatedly delete and recreate the package
- Package in "symlink mode" doesn't cause any maintenance work
- "wine-lol" actually keeps installed so future "patch requirements" can be "pushed" directly to the user
- A dedicated "LoL wine" does not need 64 bit! This would halve the build time.
- A 32 bit only build doesn't need two dynamic linkers which would make it possible to make wine-lol use the dynamic linker in wine-lol-glibc.

**Comment by M-Reimer** [comment]
Working on this one currently.

- Building for 32bit only works and creates a working (half the size) WINE build which successfully runs LoL for me.
- Using the dynamic linker from wine-lol-glibc works now as I can force the whole build process to be 32 bit which makes the 32 bit dynamic linker the only one I have to care about. I haven't tried that, but in theory this should allow wine-lol-glibc to be a different version than "real" glibc. We'll see if this works out with the next glibc update in Arch.
- I install the whole set of files to /opt/wine-lol
- Next step will be to create a symlink from /opt/wine-lol/bin/wine to /usr/bin/wine-lol

**Comment by M-Reimer** [comment]
Fixed now.
If "wine-lol" is ever replaced by "wine-staging" again, then this package will be updated to pull wine-staging as dependency and itself will only install one symlink from /usr/bin/wine to /opt/wine-lol/bin/wine
